src/index.ts

```diff
@@ -109,6 +112,39 @@ function getCurrentMigrationPath(parsedSettings: ParsedSettings) {
   return `${parsedSettings.migrationsFolder}/current.sql`;
 }
 
+async function _dump(parsedSettings: ParsedSettings, shadow = false) {
+  if (parsedSettings.dumpCommand) {
+    const logSuffix = shadow ? "[shadow]" : "";
+
+    // tslint:disable-next-line no-console
+    console.log(`graphile-migrate${logSuffix}: running dump`);
+
+    const connectionString = shadow
+      ? parsedSettings.shadowConnectionString
+      : parsedSettings.connectionString;
+    if (!connectionString) {
+      throw new Error("Could not determine connection string for dump");
+    }
+
+    const { stdout, stderr } = await exec(parsedSettings.dumpCommand, {
+      env: {
+        PATH: process.env.PATH,
+        DATABASE_URL: connectionString,
+      },
+      encoding: "utf8",
+      maxBuffer: 10 * 1024 * 1024,
+    });
+    if (stdout) {
+      // tslint:disable-next-line no-console
+      console.log(stdout);
+    }
+    if (stderr) {
+      // tslint:disable-next-line no-console
+      console.error(stderr);
+    }
```

Comment:
I know this is a copy of my existing code; I think my existing code is bad. We should probably just pass through stdio using `stdio: 'inherit'`, or, even better, process the stream and indent it to make it clear where the output is coming from. We'll have to be careful about ANSI codes though. That's partly why I've not done it yet.

For now, it would be best to move this to a shared function so you can just say:

```
await exec(parsedSettings.dumpCommand)
```

and I can refactor what that does at a later time.

Reply:
If we go the hooks route I don't think you'll need this code.